src/test/java/io/mathdojo/useraccountservice/TestConfig.java

```diff
@@ -0,0 +1,161 @@
+package io.mathdojo.useraccountservice;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+
+import io.mathdojo.useraccountservice.model.User;
+
+/**
+ * 
+ * Used for injecting mock mongodb repository beans into spring application context when
+ * testing with @RunWith(SpringRunner.class)
+ *
+ */
+@TestConfiguration
+public class TestConfig {
+	Map<String, User> createdUsers = new HashMap<>();
+
+	@Bean
+	MathDojoUserRepository userRepo() {
+		// configured known user
+		createdUsers.put("aKnownUser", new User("aKnownUser", false, "", "", "aKnownOrg"));
+		MathDojoUserRepository userRepo = new MathDojoUserRepository() {
+
+			@Override
+			public <S extends User> Optional<S> findOne(Example<S> example) {
+				// TODO Auto-generated method stub
+				return null;
+			}
+
+			@Override
+			public <S extends User> Page<S> findAll(Example<S> example, Pageable pageable) {
```

Comment:
I really like this approach. I do wonder whether we could make this more flexible & easier to use. Rather than a hardened hand-written config that replicates the actual bean (we had to implement all the methods) could we return a mochito version of this? That would allow us to control behaviour in the test in which it is being used. Thoughts?

Reply:
It may be possible, and I would be happy to switch to a mockito version if the need for flexibility arises but I thought that a concrete reusable mock repository fit our needs better, because the behaviour should ideally be always fixed just like the actual mongo repo object.

Reply:
From my understanding, using mockito wouldn't stop it from being a "concrete" implementation per se. It's more a general "code smell" if in one of your tests you end up having to implement all the methods for something that is supposed to be a mock of an implementation. It's not a hard stop for me, it's something I learned and figured was worth sharing.